### PR TITLE
I18 upgrade rails 6 1

### PR DESCRIPTION
**Pull request by aprilrieger**
I18 upgrade rails 6.1

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -30,15 +30,15 @@ gem 'listen', '>= 3.0.5', '< 3.2', group: [:development]
 gem 'mods'
 gem 'negative_captcha'
 gem 'oai', '>=1.1.0'
-gem 'pg', '~> 0.18'
+gem 'pg', '~> 1.1'
```
Comment (aprilrieger) [inline Gemfile:33]: Error:
[ E 2023-03-15 07:18:21.0708 215/T1w age/Cor/App/Implementation.cpp:221 ]: Could not spawn process for application /home/app/webapp: The application encountered the following error: Error loading the 'postgresql' Active Record adapter. Missing a gem it depends on? can't activate pg (~> 1.1), already activated pg-0.21.0. Make sure all dependencies are added to Gemfile. (LoadError)
oral-history-web-1  |   Error ID: e8691954
oral-history-web-1  |   Error details saved to: /tmp/passenger-error-44jDot.html

Needed to upgrade to pg (~> 1.1)

File: config/environments/development.rb [hunk 2]
```diff
@@ -51,14 +70,15 @@
...
+  # config.i18n.raise_on_missing_translations = true
+
+  # Annotate rendered view with file names.
+  # config.action_view.annotate_rendered_view_with_filenames = true
 
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   # Commented out for development purposes with Apple M1 Silicon Procesing Chips
   # config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+  config.hosts << "oralhistory.test"
```
Comment (aprilrieger) [inline config/environments/development.rb:83]: Had an error in the browser once navigatting to oralhistory.test about this issue: https://blog.saeloun.com/2019/10/31/rails-6-adds-guard-against-dns-rebinding-attacks, https://guides.rubyonrails.org/configuring.html.

So I added the below code to config/environments/development.rb:
config.hosts << "oralhistory.test"

File: config/initializers/assets.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@
 # Rails.application.config.assets.paths << Emoji.images_path
 # Add Yarn node_modules folder to the asset load path.
 Rails.application.config.assets.paths << Rails.root.join('node_modules')
-Rails.application.config.assets.precompile += %w( footer.js )
+# Rails.application.config.assets.precompile += %w( footer.js ) <---- this was removed - not sure if it should
```
Comment (aprilrieger) [inline config/initializers/assets.rb:10]: Question

File: config/initializers/new_framework_defaults_6_1.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Be sure to restart your server when you modify this file.
+# TODO: Need to comb through this and enable helpful deafults it suggests
```
Comment (aprilrieger) [inline config/initializers/new_framework_defaults_6_1.rb:2]: TODO

File: bin/yarn [hunk 5]
```diff
@@ -1,9 +1,15 @@
 #!/usr/bin/env ruby
-VENDOR_PATH = File.expand_path('..', __dir__)
-Dir.chdir(VENDOR_PATH) do
-  begin
-    exec "yarnpkg #{ARGV.join(" ")}"
-  rescue Errno::ENOENT
+APP_ROOT = File.expand_path('..', __dir__)
```
Comment (aprilrieger) [inline bin/yarn:2]: Question:

Not sure about the VENDOR_PATH removal

File: bin/yarn [hunk 6]
```diff
@@ -1,9 +1,15 @@
 #!/usr/bin/env ruby
-VENDOR_PATH = File.expand_path('..', __dir__)
```
Comment (aprilrieger) [inline bin/yarn:2]: Question:

Not sure about the VENDOR_PATH removal

File: config/cable.yml [hunk 7]
```diff
@@ -2,9 +2,9 @@ development:
   adapter: async
 
 test:
-  adapter: async
+  adapter: test
 
 production:
   adapter: redis
-  url: redis://localhost:6379/1
+  url: <%= ENV.fetch("REDIS_URL") { "redis://localhost:6379/1" } %>
```
Comment (aprilrieger) [inline config/cable.yml:9]: Do we need to do anything differently for the deploy with this updated env variable?
`REDIS_URL`

File: config/environments/development.rb [hunk 8]
```diff
@@ -1,6 +1,11 @@
+require "active_support/core_ext/integer/time"
+
 Rails.application.configure do
-    # Verifies that versions and hashed value of the package contents in the project's package.json
-  config.webpacker.check_yarn_integrity = true
+  # disable integrity check in development
+  config.webpacker.check_yarn_integrity = false
```
Comment (aprilrieger) [inline config/environments/development.rb:5]: Question:

Both were in development before upgrade:
  # disable integrity check in development
  config.webpacker.check_yarn_integrity = false

  # Verifies that versions and hashed value of the package contents in the project's package.json
  config.webpacker.check_yarn_integrity = true

Which "DO" we want?

File: config/environments/development.rb [hunk 9]
```diff
@@ -9,22 +14,27 @@
...
+  config.cache_classes = false
+
   # Do not eager load code on boot.
   config.eager_load = false
 
   # Show full error reports.
   config.consider_all_requests_local = true
-
-  # disable integrity check in development
-  config.webpacker.check_yarn_integrity = false
```
Comment (aprilrieger) [inline config/environments/development.rb:19]: Second instance of it but `false`

File: config/environments/development.rb [hunk 10]
```diff
@@ -1,6 +1,11 @@
+require "active_support/core_ext/integer/time"
+
 Rails.application.configure do
-    # Verifies that versions and hashed value of the package contents in the project's package.json
-  config.webpacker.check_yarn_integrity = true
```
Comment (aprilrieger) [inline config/environments/development.rb:3]: First instance of it but `true`

File: config/environments/production.rb [hunk 11]
```diff
@@ -17,51 +19,49 @@
...
+  # Ensures that a master key has been made available in either ENV["RAILS_MASTER_KEY"]
+  # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
+  # config.require_master_key = true
 
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
   config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
 
-  # Compress JavaScripts and CSS.
-  config.assets.js_compressor = :uglifier
```
Comment (aprilrieger) [inline config/environments/production.rb:30]: Should I carry this over?

File: config/environments/production.rb [hunk 12]
```diff
@@ -84,6 +84,7 @@
     password: ENV.fetch('SMTP_PASSWORD', nil)
   }
 
+  config.action_mailer.perform_caching = false
```
Comment (aprilrieger) [inline config/environments/production.rb:87]: Is this ok as false?

File: bin/rails [hunk 13]
```diff
@@ -1,12 +1,5 @@
 #!/usr/bin/env ruby
-# Commented out for development purposes with Apple M1 Silicon Procesing Chips
-# UCLA Original: Keep -- Start
-# begin
-#   load File.expand_path('../spring', __FILE__)
-# rescue LoadError => e
-#   raise unless e.message.include?('spring')
-# end
-# UCLA Original: Keep --- End
+load File.expand_path("spring", __dir__)
```
Comment (orangewolf) [inline bin/rails:2]: you brought spring back - is that ok?

File: bin/yarn (hunk 6 above)
Comment (orangewolf) [reply bin/yarn:2]: they just renamed the variable

File: config/cable.yml (hunk 7 above)
Comment (orangewolf) [inline config/cable.yml:9]: prefer non-block version unless you need a block

```suggestion
  url: <%= ENV.fetch("REDIS_URL", "redis://localhost:6379/1") %>
```

File: config/environments/development.rb (hunk 8 above)
Comment (orangewolf) [reply config/environments/development.rb:5]: the end result would have been false before. so false makes sense

File: config/environments/development.rb (hunk 2 above)
Comment (orangewolf) [reply config/environments/development.rb:83]: this is expected

File: config/environments/development.rb [hunk 14]
```diff
@@ -51,14 +70,15 @@
   # Suppress logger output for asset requests.
   config.assets.quiet = true
 
-  # Raises error for missing translations
-  # config.action_view.raise_on_missing_translations = true
-  logger           = ActiveSupport::Logger.new(STDOUT)
-  logger.formatter = config.log_formatter
-  config.logger    = ActiveSupport::TaggedLogging.new(logger)
```
Comment (orangewolf) [inline config/environments/development.rb:58]: logging stuff is gone?

File: config/environments/production.rb (hunk 11 above)
Comment (orangewolf) [reply config/environments/production.rb:30]: shouldn't be needed I dont think. does the JS render?

File: config/environments/production.rb [hunk 15]
```diff
@@ -17,51 +19,49 @@
...
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
 
-  # Use the lowest log level to ensure availability of diagnostic information
-  # when problems arise.
-  config.log_level = :warn
+  # Include generic and useful information about system operation, but avoid logging too much
+  # information to avoid inadvertent exposure of personally identifiable information (PII).
+  config.log_level = :info
```
Comment (orangewolf) [inline config/environments/production.rb:53]: should be warn for prod

File: config/environments/production.rb (hunk 12 above)
Comment (orangewolf) [reply config/environments/production.rb:87]: yes

File: config/initializers/assets.rb (hunk 3 above)
Comment (orangewolf) [reply config/initializers/assets.rb:10]: yes it is likely needed. its hard to tell in dev mode, but in prod mode you might see a 404 for the footer.js asset

File: bin/rails (hunk 13 above)
Comment (aprilrieger) [reply bin/rails:2]: https://github.com/scientist-softserv/oral-history/pull/26/commits/fd543595260794a08a17d5dd3c9ed8b711330337

File: config/environments/development.rb (hunk 14 above)
Comment (aprilrieger) [reply config/environments/development.rb:58]: @aprilrieger Rez back

File: config/environments/production.rb (hunk 15 above)
Comment (aprilrieger) [reply config/environments/production.rb:53]: @aprilrieger Rez back

File: bin/rails (hunk 13 above)
Comment (aprilrieger) [reply bin/rails:2]: @aprilrieger Let's do some more testing around this.

File: config/environments/production.rb (hunk 11 above)
Comment (aprilrieger) [reply config/environments/production.rb:30]: Yes the JS "Does" render. I can see the progress bar js and the ajax script work for when an importer is currently running in the bg job.

Comment (aprilrieger) [reply config/environments/production.rb:30]: @aprilrieger Let's do some more testing around this

File: config/environments/development.rb (hunk 8 above)
Comment (aprilrieger) [reply config/environments/development.rb:5]: @aprilrieger Rez back FALSE

File: config/initializers/assets.rb (hunk 3 above)
Comment (aprilrieger) [reply config/initializers/assets.rb:10]: @aprilrieger rez back

File: config/environments/production.rb (hunk 11 above)
Comment (aprilrieger) [reply config/environments/production.rb:30]: This is a production file, so me testing locally isn't going to be sufficient. I am sticking it back in for now.

**Pull request closed without merging**